assets/tests/components/detours/diversionPage.test.tsx

```diff
@@ -85,10 +90,37 @@ describe("DiversionPage", () => {
       fireEvent.click(originalRouteShape.get(container))
     })
 
-    expect(await screen.findByTitle("Detour Start")).not.toBeNull()
+    expect(await screen.findByTitle("Detour Start")).toBeVisible()
     expect(screen.queryByTitle("Detour End")).not.toBeInTheDocument()
   })
 
+  test("directions start with origin intersection when second waypoint is added", async () => {
+    jest.mocked(fetchDetourDirections).mockResolvedValue(
+      Ok(
+        detourShapeFactory.build({
+          directions: [
+            { instruction: "Turn left on Main Street" },
+            { instruction: "Turn right on High Street" },
+            { instruction: "Turn sharp right on Broadway" },
+          ],
+        })
+      )
+    )
+
+    const intersection = "Avenue 1 & Street 2"
+    jest.mocked(useNearestIntersection).mockReturnValue({
+      ok: intersection,
+    })
+
+    const { container } = render(<DiversionPage />)
+
+    fireEvent.click(originalRouteShape.get(container))
+
+    fireEvent.click(originalRouteShape.get(container))
+
```

Comment:
issue: I think both of these need `act`?
```suggestion
    act(() => fireEvent.click(originalRouteShape.get(container)))

    act(() => fireEvent.click(originalRouteShape.get(container)))
```

Reply:
Ahhhh I see! I needed to read up, and [this example](https://github.com/threepointone/react-act-examples/blob/master/sync.md#events) helped it connect for me.